dnf.spec

```diff
@@ -1,5 +1,8 @@
 # Always build out-of-source
-%undefine __cmake_in_source_build
+%if (0%{?fedora} && 0%{?fedora} <= 32) || 0%{?rhel}
+%else
+    %undefine __cmake_in_source_build
+%endif
```

Comment:
I would just add `%define __cmake_in_source_build 1` here. This makes sure the build macros behave the same on all Fedoras and RHELs, forcing the old behavior.